web/controllers/game_controller.ex

```diff
@@ -0,0 +1,11 @@
+defmodule Yggdrasil.GameController do
+  use Yggdrasil.Web, :controller
+
+  alias Yggdrasil.Game
+
+  def index(conn, _params) do
+    games = Repo.all Game
```

Comment:
Also it maybe easier for me to leave it as is finish this. Them once merged
to branch off and mock up what I'm thinking. It maybe easier to see and us
to discuss so I have something to point at.

That way if we say no then we still have this and can kill the other
branch.
On Thu, Feb 18, 2016 at 11:04 PM Matthew Philyaw <matthew.philyaw@gmail.com>
wrote:

> So I see Yggdrasil encompassing the game engine. The otp app to me isn't
> just for code organization it symbolizes an independent running app.
>
> I don't see it containing a user though. I think YggdrasilWeb should
> contain the user and all administrative stuff. Possibly even having its own
> repo.
>
> So if we do this we achieve two isolated apps where the web interacts with
> game engine through a module but doesn't know or care about its internals.
>
> For example a character is something that I would say is a game engine
> component. Lots of things in the game engine will depend on it and use it.
> Thus to me it should own it and mange its life cycle.
>
> To your point yes the web exposes an API to remove a character but I'll be
> shocked if removing a character will not need some sort of logic wrapped
> around it to ensure its not in an active room this being used if so either
> error or notify the room. In addition there maybe other clean up and checks.
>
> So if the web API handles this in the controller for example then the
> controller has to know all this logic if it's going to directly manipulate
> the repo. This also means tests for ensuring the stability on the running
> game character are now in the controller. Because the game engine app can't
> test it. It has no idea a character will be removed.
>
> However if this is in an API like I suggested above Yggdrasil can ensure a
> vet the management of a characters life cycle that it depends on. Makes it
> easier to test I think. It keeps the test local to the pieces that depend
> on it heavily.
>
> It's my view anyway kind of how I have this split up. I think it's best to
> think of otp apps as black boxes that expose its functionality.
> On Thu, Feb 18, 2016 at 10:24 PM Sean Copenhaver <notifications@github.com>
> wrote:
>
>> In web/controllers/game_controller.ex
>> <https://github.com/voluspa/yggdrasil/pull/23#discussion_r53418983>:
>>
>> > @@ -0,0 +1,11 @@
>> > +defmodule Yggdrasil.GameController do
>> > +  use Yggdrasil.Web, :controller
>> > +
>> > +  alias Yggdrasil.Game
>> > +
>> > +  def index(conn, _params) do
>> > +    games = Repo.all Game
>>
>> Yeah but the game engine will never need to worry about removing
>> characters and such, that'll be a web thing. Also the web will need
>> abilities to search but the game engine will most likely only perform id
>> based lookups. Other administration things like deactivating old accounts
>> and such, the game engine doesn't care about any of that.
>>
>> The issue with a character requiring updating multiple tables might be
>> handled for us when Ecto hits 2.0, but either way I get that we might need
>> to write a module to hide that stuff behind. I do question if everything
>> the web API will need should be in Yggdrasil. So maybe I should ask do
>> you see Yggdrasil encompassing everything?
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/voluspa/yggdrasil/pull/23/files#r53418983>.
>>
>